planner/core/planbuilder.go

```diff
@@ -2022,7 +2180,7 @@ func (b *PlanBuilder) buildAnalyzeTable(as *ast.AnalyzeTableStmt, opts map[ast.A
 			}
 		}
 		if version == statistics.Version2 {
-			p.ColTasks, err = b.buildAnalyzeFullSamplingTask(as, p.ColTasks, physicalIDs, names, tbl, version)
+			p.ColTasks, p.V2AnalyzeOptions, err = b.buildAnalyzeFullSamplingTask(as, p.ColTasks, physicalIDs, names, tbl, version)
```

Comment:
Here we are in the loop of `range as.TableNames`. We add `AnalyzeColumnsTask` for different tables together into `Analyze.ColTasks`. But here we reset `Analyze.V2AnalyzeOptions` each time when dealing with a table, whose `PhyTableID` is the table ID.

Reply:
Thx for finding. Missed the cases of analyze multiple tables. Updated.